packages/canary-react/src/stories/ic-select_(multi).stories.mdx

```diff
@@ -0,0 +1,896 @@
+          console.log(`icOptionSelect: ${event.detail.value}`)
+        }
+        onIcOptionDeselect={(event) =>
+          console.log(`icOptionDeselect: ${event.detail.value}`)
+        }
+      />
+    </div>
+  </Story>
+</Canvas>
+
+#### Default code example
+
+```jsx
+import * as React from "react";
+import { IcSelectWithMulti } from "@ukic/canary-react";
+
+const options: IcMenuOption[] = [
+  { label: "Cappuccino", value: "Cap" },
+  { label: "Latte", value: "Lat" },
+  { label: "Americano", value: "Ame" },
+  { label: "Filter", value: "Fil" },
+  { label: "Flat white", value: "Fla" },
+  { label: "Mocha", value: "Moc" },
+  { label: "Macchiato", value: "Mac" },
+];
+
+const MultiSelect = () => (
+  <IcSelectWithMulti
+    label="What are your favourite types of coffee?"
+    options={options}
+    multiple
+    onIcChange={(event) => console.log(`icChange: ${event.detail.value}`)}
+    onIcOptionSelect={(event) =>
+      console.log(`icOptionSelect: ${event.detail.value}`)
+    }
+    onIcOptionDeselect={(event) =>
+      console.log(`icOptionDeselect: ${event.detail.value}`)
+    }
+  />
+);
+
+export default MultiSelect;
+```
+
+### Default value
+
+Show options as pre-selected by setting the `value` prop to an array containing the values of these options.
+
+<Canvas withSource="none">
+  <Story name="Default Value">
+    <div style={{ height: "430px" }}>
+      <IcSelectWithMulti
+        label="What are your favourite types of coffee?"
+        options={OPTIONS}
+        multiple
+        value={["Ame", "Fil", "Moc"]}
+        onIcChange={(event) => console.log(`icChange: ${event.detail.value}`)}
+      />
+    </div>
+  </Story>
+</Canvas>
+
+#### Default value code example
+
+```jsx
+import * as React from "react";
+import { IcSelectWithMulti } from "@ukic/canary-react";
+
+const options: IcMenuOption[] = [
+  { label: "Cappuccino", value: "Cap" },
+  { label: "Latte", value: "Lat" },
+  { label: "Americano", value: "Ame" },
+  { label: "Filter", value: "Fil" },
+  { label: "Flat white", value: "Fla" },
+  { label: "Mocha", value: "Moc" },
+  { label: "Macchiato", value: "Mac" },
+];
+
+const MultiSelect = () => (
+  <IcSelectWithMulti
+    label="What are your favourite types of coffee?"
+    options={options}
+    multiple
+    value={["Ame", "Fil", "Moc"]}
+    onIcChange={(event) => console.log(`icChange: ${event.detail.value}`)}
+  />
+);
+
+export default MultiSelect;
+```
+
+### With clear button
+
+Display a clear button by using the `showClearButton` prop. 
+This will appear when the user has selected some options and allow them to easily clear their selection.
+
+<Canvas withSource="none">
+  <Story name="With Clear Button">
+    <div style={{ height: "430px" }}>
+      <IcSelectWithMulti
+        label="What are your favourite types of coffee?"
+        options={OPTIONS}
+        multiple
+        showClearButton
+        onIcChange={(event) => console.log(`icChange: ${event.detail.value}`)}
```

Comment:
I first tried to make the clear button set the value to `null`, which was turning out to be not as simple as I thought it would be! But had a look at the search bar component and the clear button on that sets it to an empty string, so have decided to do the same with the clear button on the multi-select for consistency.

Reply:
I first tried to make the "Clear all" button / deselecting options set the value to `null`, which was turning out to be not as simple as I thought it would be! But had a look at the search bar component and the clear button on that sets it to an empty string, so have decided to do the same with the clear button on the multi-select for consistency.

Reply:
All three methods of clearing the selection (clear button, "Clear all" button, and deselecting individual options) now set the value to null